docs/Specifications/Workgroup Creation & Management/Interfaces and schemas for the main logical input objects and resolvers.md

```diff
@@ -0,0 +1,233 @@
+
+[https://github.com/LimeChain/etherlime/blob/446b411961c50c36797727fea725d1bdc5a962c5/packages/etherlime-utils/utils/linking-utils.js#L5](https://github.com/LimeChain/etherlime/blob/446b411961c50c36797727fea725d1bdc5a962c5/packages/etherlime-utils/utils/linking-utils.js#L5)
+
+## Contract artifacts resolver interface
+
+The contract artefacts resolver defines an abstract interface to be implemented. This interface governs how contracts are to be inputted in the baseline tooling and what the expected outputs are.
+
+Any contract artefacts resolver should be a class that adheres to the following interface:
+
+    interface ContractArtifactsResolver {
+    	resolve(contractName, extraParams) : Promise<ContractArtifactsSchema>
+    }
+
+Invocation of the `resolve` method of the resolver should return a promise returning an object adhering to `ContractArtifactsSchema` outlined in the previous section.
+
+The `contractName` param is a param allowing for the tooling to specify for which contracts it needs the artifacts to be resolved. For example, the baseline deployment tools might request the resolving of the `OrgRegistry` contract artifacts - the result of the compilation of the `OrgRegistry.sol` contract.
+
+The `extraParams` are additional resolver-specific customization params passed as an input that might be needed for the resolver to work. The idea behind `extraParams` is to allow for further customizations of the resolver behavior. An example of extra params might be authentication parameters for an FTP access to a directory containing the contract artifacts.
+
+## Inbuilt Baseline contract artifacts resolvers
+
+### ContractArtifactsPathResolver
+
+`ContractArtifactsPathResolver` is to be developed. This resolver would adhere to the `ContractArtifactsResolver` interface.
+
+    class ContractArtifactsPathResolver {
+    	constructor(path) { ... }
+    	resolve(contractName, extraParams) { ... }
+    }
+
+The `path` param is resolver specific parameter to allow the resolver to find and resolve the input data to the resulting object. The `path` param of this resolver needs to be a file system path to a directory. The `contractName` would be the name of a contract to be resolved. The resolver will look for a file named `contractName` with the extension of `.json` (as this is the common output of the compilers) in the `path` directory. Upon finding this file it will parse the object and return it to the tooling. 
+
+The `extraParams` param is unused for this resolver.
+
+## Example of a custom resolvers
+
+An example of custom `ContractArtifactsResolver` is to be developed as `Radish34ContractArtifactsPathResolver`. As Radish34 uses "0x" compiler that has a different output than Truffle and Etherlime, further modification needs to be made to the resulting bytecode in order for it to adhere to the  `ContractArtifactsSchema`. 
+
+# Verifier Key
```

Comment:
Should we consolidate this and the keystore as described above into a common "key manager" interface?

Reply:
I think that they don't share too much in common. I think the keystone is more to do with the deployed wallet, while the verifier key is to do with the zip circuits.

Reply:
I feel they share quite a bit in common, i.e., both are elliptic curve keypairs and we use both for sign/verify...